uarray/backend.py

```diff
@@ -211,40 +426,75 @@ def usable(self, dispatchable_args: Tuple["DispatchableInstance"]) -> Optional[b
 _skipped_backend: ContextVar[Set[Backend]] = ContextVar('_skipped_backend', default=set())
 
 
-class _SetBackend:
+class set_backend:
+    """
+    A context manager that sets the preferred backend.
+
+    Parameters
+    ----------
+    backend : Backend
+        The backend to set.
+    coerce: Optional[bool], optional
+        Whether to coerce the input arguments, where ``None`` lets this
+        backend be used as a fallback with coercion on failure.
+
+    See Also
+    --------
+    skip_backend: A context manager that allows skipping of backends.
+    DispatchableInstance: Items to be coerced must be marked by a DispatchableInstance.
+    """
     def __init__(self, backend: Backend, coerce: Optional[bool] = None):
-        self.token = _preferred_backend.set(((backend, coerce),) + _preferred_backend.get())
+        self.token = None
+        self.backend = backend
+        self.coerce = coerce
 
     def __enter__(self):
-        pass
+        self.token = _preferred_backend.set(((self.backend, self.coerce),) + _preferred_backend.get())
 
     def __exit__(self, exception_type, exception_value, traceback):
         _preferred_backend.reset(self.token)
 
 
-class _SkipBackend:
+class skip_backend:
+    """
+    Provides the ability to skip the preferred backend in a thread-safe manner.
+
+    When combined with :obj:`set_backend`, this context manager takes precedence.
+
+    Parameters
+    ----------
+    backend : Backend
+        The backend to skip.
+
+    See Also
+    --------
+    set_backend: A context manager that allows setting of preferred backend.
+    """
     def __init__(self, backend: Backend):
-        new_skipped = set(_skipped_backend.get())
-        new_skipped.add(backend)
-        self.token = _skipped_backend.set(new_skipped)
+        self.backend = backend
+        self.token = None
 
     def __enter__(self):
-        pass
+        new_skipped = set(_skipped_backend.get())
+        new_skipped.add(self.backend)
+        self.token = _skipped_backend.set(new_skipped)
 
     def __exit__(self, exception_type, exception_value, traceback):
         _skipped_backend.reset(self.token)
 
 
-set_backend = _SetBackend
-skip_backend = _SkipBackend
-
-
-def multimethod(backend: Backend, method: MultiMethod):
+def register_implementation(backend: Backend, method: MultiMethod):
+    """
+    Create an implementation for a given backend/method. The implementation
+    should have the same signature as the method, or perhaps with some optional
+    keyword arguments missing.
```

Comment:
only keyword arguments, so `f(a, b, *args)` is not supported? otherwise, drop "keyword"